Crating scenes and sources

##### Issue type
Feature request

##### Description
Hey there, currently working on a small scheduler/auto-pilot for obs, and was planning on using this solution for remote control, but was surprised by there not being functions to create scenes and sources.

The option is clearly there in ` libobs/obs-source.h` and `obs.h` as i could tell from taking a glance at OBS docs.

I noticed that in this project you only import frontend and module api. Is there a reason behind that, or am i missing something?

Also if that's a thing worth doing from your standpoint, i'd be happy to investigate and try write this.

Hey there, i know its kinda offtop, but i seem to be unable to resolve this issue myself.
I've set everything up as is described in instructions, and i even got obs itself to build without any problems and it lauches and workes without issues.

however when i try to configure cmake for obs-websockets, i get this error, even though i set path to the libobs
```
Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
CMake Error at CMakeLists.txt:18 (find_package):
  By not providing "FindLibObs.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "LibObs", but
  CMake did not find one.

  Could not find a package configuration file provided by "LibObs" with any
  of the following names:

    LibObsConfig.cmake
    libobs-config.cmake

  Add the installation prefix of "LibObs" to CMAKE_PREFIX_PATH or set
  "LibObs_DIR" to a directory containing one of the above files.  If "LibObs"
  provides a separate development package or SDK, be sure it has been
  installed.


Configuring incomplete, errors occurred!
See also "C:/Users/Alex/Documents/Projects/obs-websocket/obs-websocket/build/CMakeFiles/CMakeOutput.log".
```
i also tried doing it on my ubuntu machine and  got the exact same problem there. Any help with that?

--- a/src/WSRequestHandler.cpp
+++ b/src/WSRequestHandler.cpp
@@ -46,6 +46,7 @@ const QHash<QString, RpcMethodHandler> WSRequestHandler::messageMap {
 	{ "SetCurrentScene", &WSRequestHandler::SetCurrentScene },
 	{ "GetCurrentScene", &WSRequestHandler::GetCurrentScene },
 	{ "GetSceneList", &WSRequestHandler::GetSceneList },
+	{ "CreateScene", &WSRequestHandler::CreateScene },
 	{ "SetSceneTransitionOverride", &WSRequestHandler::SetSceneTransitionOverride },
 	{ "RemoveSceneTransitionOverride", &WSRequestHandler::RemoveSceneTransitionOverride },
 	{ "GetSceneTransitionOverride", &WSRequestHandler::GetSceneTransitionOverride },

--- a/src/WSRequestHandler.h
+++ b/src/WSRequestHandler.h
@@ -64,6 +64,7 @@ class WSRequestHandler {
 		RpcResponse SetCurrentScene(const RpcRequest&);
 		RpcResponse GetCurrentScene(const RpcRequest&);
 		RpcResponse GetSceneList(const RpcRequest&);
+		RpcResponse CreateScene(const RpcRequest&);
 		RpcResponse SetSceneTransitionOverride(const RpcRequest&);
 		RpcResponse RemoveSceneTransitionOverride(const RpcRequest&);
 		RpcResponse GetSceneTransitionOverride(const RpcRequest&);

--- a/src/WSRequestHandler_Scenes.cpp
+++ b/src/WSRequestHandler_Scenes.cpp
@@ -80,6 +80,31 @@ RpcResponse WSRequestHandler::GetSceneList(const RpcRequest& request) {
 }
 
 /**
+ * Create a new scene scene.
+ *
+ * @param {String} `scene-name` Name of the scene to create.
+ *
+ * @api requests
+ * @name CreateScene
+ * @category scenes
+ * @since 4.8.0
+ */
+RpcResponse WSRequestHandler::CreateScene(const RpcRequest& request) {
+	if (!request.hasField("sceneName")) {
+		return request.failed("missing request parameters");
+	}
+
+	const char* sceneName = obs_data_get_string(request.parameters(), "sceneName");
+	OBSSourceAutoRelease source = obs_get_source_by_name(sceneName);
+
+	if (source) {
+		return request.failed("scene with this name already exists");
+	}
+	obs_scene_create(sceneName);
+	return request.success();
+}
+
+/**
 * Changes the order of scene items in the requested scene.
 *
 * @param {String (optional)} `scene` Name of the scene to reorder (defaults to current).

--- a/src/WSRequestHandler_Scenes.cpp
+++ b/src/WSRequestHandler_Scenes.cpp
@@ -82,7 +82,7 @@ RpcResponse WSRequestHandler::GetSceneList(const RpcRequest& request) {
 /**
  * Create a new scene scene.
  *
- * @param {String} `scene-name` Name of the scene to create.
+ * @param {String} `sceneName` Name of the scene to create.
  *
  * @api requests
  * @name CreateScene

